cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/flow/BasicFlowletContext.java

```diff
@@ -83,6 +86,18 @@ public MetricsCollector load(String key) throws Exception {
         }
       });
 
+    this.producerMetrics = CacheBuilder.newBuilder()
```

Comment:
why do you need a cache? MetricsCollectionService has a cache for collectors, the only difference is there the expiry time is for a minute, while its an hour here. is that why you added this cache - longer time for expiry ?